src/functions-and-arrays.js

```diff
@@ -1,41 +1,123 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
+function maxOfTwoNumbers(numb1, numb2) {
+  if (numb1 > numb2) {
+    return numb1;
+  } else if (numb1 < numb2) {
+    return numb2;
+  } else if (numb1 === numb2) {
+    return numb1 || numb2
+  } else {
+    return console.log("Could not calculate the maximum of two numbers")
+  }
+}
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
-
+function findLongestWord(wordsArray) {
+  let longestWord = "";
+  if (!wordsArray.length) return null;
+  wordsArray.forEach((word) => {
+    if (word.length > longestWord.length) {
+      longestWord = word;
+    }
+  })
+  return longestWord;
+}
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbersArray) {
+  let sum = 0;
+  if (!numbersArray.length) return 0;
+  numbersArray.forEach((number) => {
+    sum += number;
+  })
+  return sum;
+}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(mixedArray) {
+  let sum = 0;
+    if (!mixedArray.length) return sum;
+    mixedArray.forEach((value) => {
+      if (typeof value === "number") {
+        sum += value;
+      } else if (typeof value === "boolean") {
+        if (value === true) {
+          sum += 1;
+        }
+      } else if (typeof value === "string") {
+        sum += value.length;
+      } else {
+        throw new Error("Unsupported Data Type found in Array");
+      }
+    })
+    return sum;
+}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function sumNumbersArray(numbersArray) {
+  let sum = 0;
+  if (!numbersArray.length) return sum;
+  numbersArray.forEach((number) => {
+    sum += number;
+  })
+  return sum;
+}
+
+function averageNumbers(numbersArray) {
+  if (!numbersArray.length) return null;
+  let average = sumNumbersArray(numbersArray) / numbersArray.length;
+  return average;
+}
 
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function sumAllWords(wordsArray) {
+  sumOfWords = 0;
+  wordsArray.forEach((word) => {
+    sumOfWords += word.length;
+  })
+  return sumOfWords;
+}
+
+function averageWordLength(wordsArray) {
+  if (!wordsArray.length) return null;
+  let averageWordLength = sumAllWords(wordsArray) / wordsArray.length
+  return averageWordLength;
+}
 
 // Bonus - Iteration #4.1
-function avg() {}
+function avg(mixArr) {
+  if (!mixArr.length) return null;
```

Comment:
As you had already done the `function sum()` you could have reused it in here, just:

function avg (mixArr) {
  if (mixArr.length === 0) return null;
  else return sum(mixArr) / mixArr.length;
}